src/sentry/debug_files/artifact_bundle_indexing.py

```diff
@@ -216,21 +230,81 @@ def mark_bundle_for_flat_file_indexing(
 
 @sentry_sdk.tracing.trace
 def remove_artifact_bundle_from_indexes(artifact_bundle: ArtifactBundle):
+    redis_client = get_redis_cluster_for_artifact_bundles()
+
     flat_file_indexes = ArtifactBundleFlatFileIndex.objects.filter(
         flatfileindexstate__artifact_bundle=artifact_bundle
     )
     for idx in flat_file_indexes:
-        identifier = FlatFileIdentifier(
-            project_id=idx.project_id, release=idx.release_name, dist=idx.dist_name
-        )
+        identifier = FlatFileIdentifier.from_index(idx)
 
         was_removed = update_artifact_bundle_index(
             identifier, bundles_to_remove=[artifact_bundle.id]
         )
         if not was_removed:
             metrics.incr("artifact_bundle_flat_file_indexing.removal.would_block")
-            # TODO: mark this for async removal
-            pass
+            redis_client.sadd(get_deletion_key(idx.id), artifact_bundle.id)
+
+
+@sentry_sdk.tracing.trace
+def backfill_artifact_index_updates():
+    redis_client = get_redis_cluster_for_artifact_bundles()
+
+    indexes_needing_update = list(
+        ArtifactBundleFlatFileIndex.objects.filter(
+            flatfileindexstate__indexing_state=ArtifactBundleIndexingState.NOT_INDEXED.value,
+        ).distinct()[:10]
```

Comment:
Can we lift this limit into a more explicit constant that is easier to read and find?